java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java

```diff
@@ -198,9 +180,20 @@ void release(final BufferLedger ledger) {
 
   /**
    * Return the size of underlying chunk of memory managed by this Allocation Manager.
-   * @return size of memory chunk
+   *
+   * @return size of underlying memory chunk
    */
-  public int getSize() {
+  protected int getSize() {
     return size;
   }
+
+  /**
+   * Return the absolute memory address pointing to the fist byte of underling memory chunk.
+   */
+  abstract long memoryAddress();
+
+  /**
+   * Release the underling memory chunk.
+   */
+  abstract void release0();
```

Comment:
protected?

Reply:
I was thinking to force the implementations to be packaged under `org.apache.arrow.memory`... Which doesn't seem to be necessary.

Reply:
Done. Also changed `memoryAddress()` and constructor.